include/z64actor.h

```diff
@@ -38,11 +37,45 @@ typedef struct {
     /* 0x1C */ ActorFunc draw; // Draw function
 } ActorInit; // size = 0x20
 
-typedef enum {
-    /* 0 */ ALLOCTYPE_NORMAL,
-    /* 1 */ ALLOCTYPE_ABSOLUTE,
-    /* 2 */ ALLOCTYPE_PERMANENT
-} AllocType;
+/**
+ * @see ACTOR_ALLOC_ABSOLUTE
+ */
+#define ACTOR_OVERLAY_ABSOLUTE_SPACE_SIZE 0x27A0
```

Comment:
I mourn the Eva reference.

Reply:
for a more serious comment, there should be some sort of correlation to this number and the reservation size. Ideally this shouldn't be a constant value

Reply:
Codegen won't let this not be a constant

Reply:
We have some ghastly construct in MM, let me see if I can find what we did.

Reply:
Ah, apparently we just ended up with
```c
// This value is hardcoded to be the size of ovl_Arrow_Fire which currently is the biggest actor that uses the AM_FIELD.
#define AM_FIELD_SIZE SEGMENT_SIZE(ovl_Arrow_Fire)
```

but we certainly considered using a `MAX`.

Reply:
well maybe I missed something but in oot no "absolute alloc" actor overlay is `0x27A0` bytes

also how does this even match, if this relies on symbols the codegen should be ruined?

Reply:
```c
#define SEGMENT_START(segment) (_ ## segment ## SegmentStart)
#define SEGMENT_END(segment) (_ ## segment ## SegmentEnd)
#define SEGMENT_SIZE(segment) ((uintptr_t)SEGMENT_END(segment) - (uintptr_t)SEGMENT_START(segment))
```
They're in terms of segment symbols, it might get away with it because it's just a function argument?

Reply:
> also how does this even match, if this relies on symbols the codegen should be ruined?

Yes, MM actually relies on the vram symbols and performs a subtraction: 
```mips
/* 015274 800BAD34 3C098092 */  lui         $t1, %hi(ArrowIce_SetupAction)
/* 015278 800BAD38 25292430 */  addiu       $t1, $t1, %lo(ArrowIce_SetupAction)
/* 01527C 800BAD3C 1480000A */  bnez        $a0, .L800BAD68
/* 015280 800BAD40 3C0A8092 */   lui        $t2, %hi(ArrowFire_SetupAction)
/* 015284 800BAD44 254A0340 */  addiu       $t2, $t2, %lo(ArrowFire_SetupAction)
/* 015288 800BAD48 012A2023 */  subu        $a0, $t1, $t2
/* 01528C 800BAD4C AFA30024 */  sw          $v1, 0x24($sp)
/* 015290 800BAD50 0C040B22 */  jal         ZeldaArena_MallocR
/* 015294 800BAD54 AFA70038 */   sw         $a3, 0x38($sp)
```

There's no way this could match on OoT, since it has a hardcoded value instead
```mips
/* 015264 800320C4 3C058013 */  lui         $a1, %hi(STR_801366E8)
/* 015268 800320C8 24A566E8 */  addiu       $a1, $a1, %lo(STR_801366E8)
/* 01526C 800320CC 8D8D013C */  lw          $t5, 0x13C($t4)
/* 015270 800320D0 240427A0 */  addiu       $a0, $zero, 0x27A0
/* 015274 800320D4 55A0000F */  bnel        $t5, $zero, .L80032114
/* 015278 800320D8 8FB90060 */   lw         $t9, 0x60($sp)
/* 01527C 800320DC 0C01EBE3 */  jal         ZeldaArena_MallocRDebug
/* 015280 800320E0 00003025 */   move       $a2, $zero
```

To add more weirdness to this, I checked what OoT 1.0 does, and it has a hardcoded value too, but a different one...
```mips
/* 0140FC 8002519C 14600007 */  bnez        $v1, .L800251BC
/* 014100 800251A0 00000000 */   nop
/* 014104 800251A4 0C019B0E */  jal         ZeldaArena_MallocR
/* 014108 800251A8 240424E0 */   addiu      $a0, $zero, 0x24E0
/* 01410C 800251AC 8FAC0058 */  lw          $t4, 0x58($sp)
```

Dunno why they liked to change this value so much